posturebest/ViewModel/BLEManager.swift

```diff
@@ -0,0 +1,116 @@
+//
+//  BLEManager.swift
+//  posturebest
+//
+//  Created by Kaitlyn O’Flaherty on 10/2/24.
+//
+
+import Foundation
+import CoreBluetooth
+
+class BLEManager: NSObject, ObservableObject, CBCentralManagerDelegate, CBPeripheralDelegate {
+    var myCentral: CBCentralManager!
+    @Published var isSwitchedOn = false
+    @Published var periperals = [Peripheral]() // stores discovered periphs
+    @Published var connectedPeripheralUUID: UUID?
+    var cbPeripheral: CBPeripheral?
+    
+    override init() {
+        super.init()
+        myCentral = CBCentralManager(delegate: self, queue: nil)
+    }
+    
+    // Delegate method called when state of central manager updated
+    func centralManagerDidUpdateState(_ central: CBCentralManager) {
+        isSwitchedOn = central.state == .poweredOn
+        if isSwitchedOn {
+            startScanning()
+        } else {
+            stopScanning()
+        }
+    }
+    
+    // Delegate method for when peripheral discovered
+    func centralManager(_ central: CBCentralManager, didDiscover peripheral: CBPeripheral, advertisementData: [String: Any], rssi RSSI: NSNumber) {
+        let newPeripheral = Peripheral(id: peripheral.identifier, name: peripheral.name ?? "Unknown", rssi: RSSI.intValue)
+        
+        // Add peripheral to list if not there
+        if !periperals.contains(where: {$0.id == newPeripheral.id}) {
+            DispatchQueue.main.async{
+                self.periperals.append(newPeripheral)
+            }
+        }
+    }
+    
+    func startScanning() {
+        print("Scanning for devices...")
+        // Right now not scanning for specific services... can be edited later
```

Comment:
```suggestion
       
```
I'd remove this comment. since we ended up filtering out the "Unknown"

Reply:
im actually filtering out the name "unknown" but i wrote that comment to come back to filtering by service. so if our device has specific services we can only scan for those